src/sockets/matchmaking.ts

```diff
@@ -0,0 +1,33 @@
+export class Matchmaking {
+    userQueue: string[];
+
+    constructor() {
+        this.userQueue = [];
+    }
+
+    // Add user to queue
+    addUserToQueue(username: string): void {
+        if (!(username in this.userQueue)) {
+            this.userQueue.push(username);
+        }
+    }
+
+    // Remove user from queue
+    removeUser(username: string): void {
+        const index = this.userQueue.indexOf(username);
+        if (index !== -1) {
+            this.userQueue.splice(index, 1);
+        }
+    }
+
+    // Checks if number users in queue >= specified roomSize.
+    // If yes, returns the list of users and removes them from the queue.
+    // If < specified roomSize, returns null
+    checkRoomPossible(roomSize: number): string[] | null {
```

Comment:
Let's remove the roomSize input here. If we need to configure anything for the lifetime of this object, let's do it in the constructor.

For this PR, lets just hard code to 6 players in this file.